packages/starknet-snap/src/signMessage.ts

```diff
@@ -18,27 +25,36 @@ export async function signMessage(params: ApiParams) {
     const typedDataMessage = requestParamsObj.typedDataMessage;
     const network = getNetworkFromChainId(state, requestParamsObj.chainId);
 
-    if (await isUpgradeRequired(network, signerAddress)) {
-      await showUpgradeRequestModal(wallet);
-      throw new Error('Upgrade required');
-    }
-
-    logger.log(`signMessage:\nsignerAddress: ${signerAddress}\ntypedDataMessage: ${toJson(typedDataMessage)}`);
-
+    const { privateKey: signerPrivateKey, publicKey } = await getKeysFromAddress(
```

Comment:
move this and validateAccountRequireUpgradeOrDeploy  after validateAndParseAddress